packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -683,15 +683,19 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     }());
 
     dynamic caughtException;
-    void handleError(dynamic error, StackTrace stackTrace) => caughtException ??= error;
+    StackTrace? stackTrace;
+    void handleError(dynamic error, StackTrace trace)  {
+      caughtException ??= error;
+      stackTrace ??= trace;
+    }
 
     await Future<void>.microtask(() { binding.handleBeginFrame(duration); }).catchError(handleError);
     await idle();
     await Future<void>.microtask(() { binding.handleDrawFrame(); }).catchError(handleError);
     await idle();
 
     if (caughtException != null) {
-      throw caughtException as Object; // ignore: only_throw_errors, rethrowing caught exception.
+      Error.throwWithStackTrace(caughtException as Object, stackTrace!);
```

Comment:
nice, is this api new? there's probably other places we could do this

Reply:
It was new in Dart 2.16:
https://api.dart.dev/stable/3.4.4/dart-core/Error/throwWithStackTrace.html

It is indeed a nice feature!

Reply:
I was scratching my head looking at this saying "there's got to be another way" and then I found the bug you filed, @Hixie, wanting exactly this. The VM team has the suggestion there to use this new API.

Reply:
breadcrumbs: https://github.com/dart-lang/sdk/issues/35494 -> https://github.com/dart-lang/sdk/issues/30741#issuecomment-1127514020